src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs

```diff
@@ -85,6 +85,22 @@ public static void MarshalStringToCS()
             Assert.Equal("hello", HelperMarshal._stringResource);
         }
 
+        [Fact]
+        public static void MarshalUnicodeStringToCS()
+        {
+            HelperMarshal._stringResource = null;
+            Runtime.InvokeJS("App.call_test_method(\"StoreAndReturnNew\", [' '+\"Příliš žluťoučký\u202fkůň úpěl ďábelské ódy\"])");
```

Comment:
we generally don't use unicode chars directly in the source code, see coding style rule 16: https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md

> 16. When including non-ASCII characters in the source code use Unicode escape sequences (\uXXXX) instead of literal characters. Literal non-ASCII characters occasionally get garbled by a tool or editor.